rpm/matcher.go

```diff
@@ -0,0 +1,47 @@
+package rpm
+
+import (
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+
+	version "github.com/knqyf263/go-rpm-version"
+)
+
+// Matcher implements driver.Matcher.
+type Matcher struct{}
+
+var _ driver.Matcher = (*Matcher)(nil)
+
+// Filter implements driver.Matcher.
+func (*Matcher) Filter(record *claircore.IndexRecord) bool {
+	// record.Repository.Kind != "rpm"
```

Comment:
This is wrong in the case when the record stream this Matcher is dealing with has more then just RPM records. This is an issue of having a single rpm `Matcher`. You must still filter for RPM records or you'll attempt to query non-rpm packages.

Reply:
There's no way to filter for rpm records

Reply:
@hdonnay I see.
I would create per-distro matchers in this case like the others. So a RHEL matcher which filters for something in record.Dist.